socorro/processor/rules/memory_report_extraction.py

```diff
@@ -0,0 +1,132 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from socorro.lib.transform_rules import Rule
+
+
+UNITS_BYTES = 0
+KIND_NONHEAP = 0
+KIND_HEAP = 1
+
+
+class MemoryReportExtraction(Rule):
+    """Extract key measurements from the memory_report object into a more
+    comprehensible and usable dictionary. """
+
+    def version(self):
+        return '1.0'
+
+    def _predicate(self, raw_crash, raw_dumps, processed_crash, proc_meta):
+        try:
+            return (
+                bool(processed_crash['memory_report']) and
+                'pid' in processed_crash['json_dump']
+            )
+        except KeyError:
+            return False
+
+    def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
+        pid = processed_crash['json_dump']['pid']
+        memory_report = processed_crash['memory_report']
+
+        try:
+            measures = self._get_memory_measures(memory_report, pid)
+        except ValueError as e:
```

Comment:
`_get_memory_measures` looks like it's doing a lot of dict lookups. Given we don't control what is in the memory report and it could theoretically change at any time without us knowing, I we should catch `KeyError` here, too.